API/c/Inc/delay.h

```diff
@@ -0,0 +1,35 @@
+#ifndef __DELAY_H
+#define __DELAY_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include  <unistd.h>
+
+/**
+ * Current tick in sysclock ticks
+ */
+uint32_t DWT_GetTick();
```

Comment:
I should normalize all the delays to be `HAL_Delay_XX` where `XX` is `us` or `ms`.

Clean up sys tick stuff